src/btree/bt_sync_obsolete.c

```diff
@@ -480,25 +513,51 @@ __checkpoint_cleanup_eligibility(WT_SESSION_IMPL *session, const char *uri, cons
         if (ret == 0)
             newest_stop_durable_ts = WT_MAX(newest_stop_durable_ts, (wt_timestamp_t)value.val);
         WT_RET_NOTFOUND_OK(ret);
+        ret = __wt_config_subgets(session, &cval, "newest_start_durable_ts", &value);
+        if (ret == 0)
+            newest_start_durable_ts = WT_MAX(newest_start_durable_ts, (wt_timestamp_t)value.val);
+        WT_RET_NOTFOUND_OK(ret);
+        ret = __wt_config_subgets(session, &cval, "newest_txn", &value);
+        if (ret == 0)
+            newest_txn = WT_MAX(newest_txn, (uint64_t)value.val);
+        WT_RET_NOTFOUND_OK(ret);
         ret = __wt_config_subgets(session, &cval, "addr", &value);
         if (ret == 0)
             addr_size = value.len;
         WT_RET_NOTFOUND_OK(ret);
     }
     WT_RET_NOTFOUND_OK(ret);
 
+    /* Skip an empty or newly created table. */
+    if (addr_size == 0)
+        return (false);
+
+    /* The table has a durable stop timestamp. */
+    if (newest_stop_durable_ts != WT_TS_NONE)
+        return (true);
+
+    /*
+     * Logged table. The logged tables do not support timestamps, so we need to check for obsolete
+     * pages in them.
+     */
+    if (logged)
+        return (true);
+
+    /*
+     * History store table. This table contains the historical versions that are needed to be
+     * removed regularly. This condition is required when timestamps are not in use, otherwise, the
+     * first condition will be satisfied.
+     */
+    if (strcmp(uri, WT_HS_URI) == 0)
+        return (true);
+
     /*
-     * The checkpoint cleanup eligibility is decided based on the following:
-     * 1. Not an empty or newly created table.
-     * 2. The table has a durable stop timestamp.
-     * 3. Logged table. The logged tables do not support timestamps, so we need
-     *    to check for obsolete pages in them.
-     * 4. History store table. This table contains the historical versions that
-     *    are needed to be removed regularly. This condition is required when
-     *    timestamps are not in use, otherwise, the first condition will be satisfied.
+     * The checkpoint has some obsolete time window information that is no longer required to exist
+     * in the btree. Remove the obsolete data to reduce the checkpoint size.
      */
-    if ((addr_size != 0) &&
-      (newest_stop_durable_ts != WT_TS_NONE || logged || strcmp(uri, WT_HS_URI) == 0))
+    newest_ts = WT_MAX(newest_start_durable_ts, newest_stop_durable_ts);
+    if ((newest_txn != WT_TXN_NONE || newest_ts != WT_TS_NONE) &&
```

Comment:
I'd split the statement as I suggested [here](https://github.com/wiredtiger/wiredtiger/pull/10831/files#r1671410865) but I don't have a strong opinion here because there is only one line in this block while in the other, it was a whole section.

Reply:
Fixed.